libsolidity/analysis/TypeChecker.cpp

```diff
@@ -94,6 +94,63 @@ bool TypeChecker::visit(ContractDefinition const& _contract)
 	for (auto const& n: _contract.subNodes())
 		n->accept(*this);
 
+	// Check base contracts for abi encoder compatibility
+	if (!_contract.sourceUnit().annotation().experimentalFeatures.count(ExperimentalFeature::ABIEncoderV2))
+	{
+		bool addedErrors = false;
+
+		for (auto const& base: _contract.annotation().linearizedBaseContracts)
+		{
+			SecondarySourceLocation errors;
+
+			if (base == &_contract)
+				continue;
+
+			solAssert(base, "Base contract not available.");
+
+			if (base->sourceUnit().annotation().experimentalFeatures.count(ExperimentalFeature::ABIEncoderV2))
+			{
+				for (auto const& func: base->definedFunctions())
```

Comment:
Can you just loop once over `_contract.interfaceFunctionList()`? This would reduce code duplication and would also not require looping over the bases.

Reply:
You can still get the secondary source location using `f->declaration()`.